tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -1251,5 +1251,61 @@ def test_assertion_is_failure(dut):
     assert False
 
 
+class MyException(Exception):
+    pass
+
+
+@cocotb.test(expect_error=MyException)
+def test_expect_particular_exception(dut):
+    yield Timer(1)
+    raise MyException()
+
+
+@cocotb.test(expect_error=(MyException, ValueError))
+def test_expect_exception_list(dut):
+    yield Timer(1)
+    raise MyException()
+
+
+@cocotb.coroutine
+def example():
+    yield Timer(10, 'ns')
+    raise ReturnValue(1)
+
+
+@cocotb.test()
+def test_timeout_func_fail(dut):
+    try:
+        yield cocotb.triggers.with_timeout(example(), timeout_time=1, timeout_unit='ns')
+    except cocotb.result.SimTimeoutError:
+        pass
+    else:
+        assert False, "Expected a Timeout"
+
+
+@cocotb.test()
+def test_timeout_func_pass(dut):
+    res = yield cocotb.triggers.with_timeout(example(), timeout_time=100, timeout_unit='ns')
+    assert res == 1
+
+
+@cocotb.test(expect_error=True, timeout_time=1, timeout_unit='ns')
+def test_timeout_testdec_fail(dut):
+    yield Timer(10, 'ns')
+
+
+@cocotb.test(timeout_time=100, timeout_unit='ns')
+def test_timeout_testdec_pass(dut):
+    yield Timer(10, 'ns')
+
+
+@cocotb.test(expect_error=True, timeout_time=10, timeout_unit='ns')
+def test_timeout_testdec_simultaneous(dut):
+    yield Timer(10, 'ns')
+    # Whether this test fails or passes depends on the behavior of the
+    # scheduler and the implementation of the timeout function.
+    # CAUTION: THIS MAY CHANGE
```

Comment:
Again, I'd prefer you test this with a `try`/`except`:
```python
    try:
        yield cocotb.triggers.with_timeout(Timer(1, 'ns'), timeout_time=1, timeout_unit='ns')
    except cocotb.result.SimTimeoutError:
        pass
    else:
        assert False, "Expected a Timeout"
    # Whether this test fails or passes depends on the behavior of the
    # scheduler and the implementation of the timeout function.
    # CAUTION: THIS MAY CHANGE
```